### ISSUE TITLE
pipx uninstall . removes all pipx virtual environments

### ISSUE THREAD

**Issue by mateuszowolf** [pipxproject/pipx#631]
**Describe the bug**
I was testing installation of a CLI I'm developing with `pipx install .` to pick up the setup.py file in my current directory. After successfully installing it, I wanted to make some changes so I thought I could uninstall it similarly by `pipx uninstall .`.  I thought this would similarly look for a setup.py and uninstall my code. Instead pipx removed all pipx virtual environments I had installed and greeted me with a cheerful message of 

`uninstalled venvs! ✨ 🌟 ✨`

**How to reproduce**
**I do not recommend running this** as it will remove all your pipx venvs
```
pipx uninstall --verbose .                                                                                                                                                      
pipx > (setup:567): pipx version is 0.15.6.0
pipx > (setup:568): Default python interpreter is '/Users/mateusz/.pyenv/versions/3.8.6/bin/python3.8'
pipx > (run_pipx_command:135): Virtual Environment location is /Users/mateusz/.local/pipx/venvs
pipx > (needs_upgrade:54): Time since last upgrade of shared libs, in seconds: 970770.1499598026. Upgrade will be run by pipx if greater than 2592000.0.
pipx > (rmdir:17): removing directory /Users/mateusz/.local/pipx/venvs
uninstalled venvs! ✨ 🌟 ✨
```

**Expected behavior**
At least a warning about such a destructive change

**Comment by itsayellow** [comment]
Oh wowwww.  That is not expected [UNDERSTATEMENT].  Thanks for bringing this to our attention!

**Comment by itsayellow** [comment]
OK, I'm going to close this issue because due to a change present in released version 0.16.0.0, pipx no longer exhibits this bug.

Upgrading to 0.16.0.0 should prevent this from ever happening to you again:
```shell
> pipx install pylint
  installed package pylint 2.6.2, Python 3.9.1
  These apps are now globally available
    - epylint
    - pylint
    - pyreverse
    - symilar
done! ✨ 🌟 ✨
> pipx install flake8
  installed package flake8 3.8.4, Python 3.9.1
  These apps are now globally available
    - flake8
done! ✨ 🌟 ✨
> pipx list
venvs are in /Users/mclapp/.local/pipx/venvs
apps are exposed on your $PATH at /Users/mclapp/.local/bin
   package flake8 3.8.4, Python 3.9.1
    - flake8
   package pylint 2.6.2, Python 3.9.1
    - epylint
    - pylint
    - pyreverse
    - symilar
> pipx uninstall .
Nothing to uninstall for - 😴
> pipx --version
0.16.0.0
```

The relevant line that allows this bug to happen in 0.15.6.0:
https://github.com/pipxproject/pipx/blob/0a04b4847c2fa0370b33d9c959376a96b2b9fdbd/src/pipx/venv.py#L59
That same line (fixed) in 0.16.0.0:
https://github.com/pipxproject/pipx/blob/83c996111f3708666a55131d5805ec268628ceaa/src/pipx/venv.py#L61

Canonicalizing the name will turn `.` or `..` (or any number of periods) into one `-`.  This will prevent it from being interpreted as a directory name.
